crates/miden-objects/src/ecdsa_signer.rs

```diff
@@ -0,0 +1,46 @@
+use miden_core::Word;
+
+use crate::crypto::dsa::ecdsa_k256_keccak as ecdsa;
+
+// ECDSA SIGNER
+// ================================================================================================
+
+pub trait EcdsaSigner {
+    fn sign(&self, message: Word) -> ecdsa::Signature;
+    fn public_key(&self) -> ecdsa::PublicKey;
+}
```

Comment:
I think we should add docs for the type and the functions. The functions could be basic, but for the type, I think it would be good to explain its main purpose in a few words.

Separately, why do we need this trait and cannot just use `SecretKey` directly? Where else is this going to be implemented?